### ISSUE TITLE
resolve argument "colors" to "color" argument

### ISSUE THREAD

**Issue by swreeser** [webpack/webpack#2786]
**I'm submitting a feature request**
`webpack --colors` should work the same as `webpack --color`.
It changed when switching to yargs, looks like an accidental change.
@sokra more pretty CLI options, added --verbose option https://github.com/webpack/webpack/commit/2e5a5f9db3c1f3179a77eb625c67235bf359eabb ([before](https://github.com/webpack/webpack/blob/46b12b2095577391fbd88b6d29b7f0b858db4c5c/bin/webpack.js#L23)...[after](https://github.com/webpack/webpack/blob/2e5a5f9db3c1f3179a77eb625c67235bf359eabb/bin/webpack.js#L31-L35))

webpack 2.1.0-beta.15
Windows 10

**Current behavior:**
`webpack --color` uses webpack.config.js and packs my project displaying color on the terminal
`webpack` packs my project, ts-loader still outputs one line in color
`webpack --colors` shows help and usage information
along with error "Unknown argument: colors"

**Expected/desired behavior:**
`webpack --color` shows same as `webpack --colors`

**Reproduction** 
Clone the angular2 webpack starter project here
`git clone git@github.com:AngularClass/angular2-webpack-starter.git`
At this point `npm run build` should work just fine...

Update webpack

```
npm uninstall webpack
npm install webpack@2.1.0-beta.15 -S -E
```

Install dependencies and run the build

```
npm i
npm run build
```

**What is the expected behavior?**
`--colors` === `--color`

**What is the motivation / use case for changing the behavior?**
When I upgrade webpack I should not have to change `webpack --colors` to `webpack --color`. It's easy to work around and webpack provides meaningful output but it shouldn't get in the way of upgrading to webpack 2.

**Comment by TheLarkInn** [comment]
Will gladly accept a PR for this @swreeser. If you are willing I can help point you in the right direction.

**Comment by sokra** [comment]
We actually don't handle `--color` ourselves. It's handled by https://github.com/chalk/supports-color

> It obeys the --color and --no-color CLI flags.

So it was always `--color`, `--colors` never worked. It was only documented wrongly...

**Comment by TheLarkInn** [comment]
Awesome!!!! @swreeser does this answer the question?

**Comment by swreeser** [comment]
Thank you both! That doesn't quite answer my question. Supports-color treats --color and --colors the same, so why doesn't --colors work? Why doesn't --no-color work?

## Is yargs.strict() necessary?
Supports-color [has tests for --no-color and --colors ](https://github.com/chalk/supports-color/blob/master/test.js#L24-L48)

Supports-color arguments like --no-color and are not supported because yargs.strict(). [bin/config-yargs.js:273](https://github.com/webpack/webpack/blob/master/bin/config-yargs.js#L273)

**If strict() was removed then supports-color would work as expected.**

If it weren't for the argument [--color being configured for yargs](https://github.com/webpack/webpack/blob/master/bin/webpack.js#L31) then --color would through an error too.

**Comment by sokra** [comment]
> Is yargs.strict() necessary?

no, but I want it to prevent passing custom arguments to webpack. Every custom argument should use the `--env` mechanism.

We must add `--no-color` to the yargs configuration. PR?

**Comment by swreeser** [comment]
Pull Request adds colors as an alias and includes a default value function that uses supports-color.

This line may be irrelevant now:
https://github.com/webpack/webpack/blob/master/bin/webpack.js#L164-L165
But I didn't remove it.

**Comment by SpaceK33z** [comment]
Closing since the PR was merged (#2788).
